third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -420,62 +820,136 @@ impl<'a> FunctionGenerator<'a> {
                 self.gen_invoke(ctx, dest, source);
             },
             Operation::Pack(mid, sid, inst) => {
-                self.gen_struct_oper(
-                    ctx,
-                    dest,
-                    mid.qualified(*sid),
-                    inst,
-                    source,
-                    FF::Bytecode::Pack,
-                    FF::Bytecode::PackGeneric,
-                );
+                let struct_env = &fun_ctx.module.env.get_struct(mid.qualified(*sid));
+                let fun_mid = ctx.fun_ctx.fun.func_env.module_env.get_id();
+                if mid != &fun_mid
+                    && struct_env
+                        .env()
+                        .language_version()
+                        .language_version_for_public_struct()
+                {
+                    self.gen_pack_api_call(ctx, dest, source, fun_ctx, &None, inst, struct_env);
+                } else {
```

Comment:
The `else` here could also mean lang version <= 2.3 and cross module. But that would be an error caught earlier and we would never reach that condition here?

Assuming this, IMO the check for language version better serves as an `assert!` instead of being lumped in together with the cross module check.

Reply:
same applies elsewhere in this file.

Reply:
changed to internal error in the if branch